### DISCUSSION TITLE
[Art competition] Program analysis

### DISCUSSION THREAD

**Discussion by p51lee** [prosyslab-classroom/cs524-program-analysis#268] [Art competition]
![DALL·E 2022-11-16 10 40 41 - Me and my opponent playing poker, Paul Cézanne](https://user-images.githubusercontent.com/68288688/202062746-6663c385-dcac-4560-8c75-8b04b3818b17.png)

## Short Description
For me, [*program analysis*] means [*Me and my opponent playing poker, Paul Cézanne*]".

## Long Description (up to 3 sentences)
Program analysis (static analysis) computes undecidable semantic properties by approximating them through abstraction.
In poker, you can never directly know your opponent's hand. It is very similar to static analysis in that it makes the best choice by abstracting the opponent's hand through the information of my hand and the rules of the game.

**Comment by KihongHeo** [comment]
Good insight.
